### PR TITLE
chore: split out e2e tests MONGOSH-1251

### PR DESCRIPTION
**Pull request by lerouxb**

### CODE REVIEW

File: packages/e2e-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# @mongosh/e2e-tests
+
+E2E tests for Mongosh
```
Comment (addaleax) [inline packages/e2e-tests/README.md:3]:
```suggestion
E2E tests for mongosh
```

File: packages/e2e-tests/bin/mongosh.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env node
+require('../lib/run');
```
Comment (addaleax) [inline packages/e2e-tests/bin/mongosh.js:2]: Doesn’t make sense to keep this file, no?

File: packages/e2e-tests/package.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
+{
+  "name": "@mongosh/e2e-tests",
+  "version": "0.0.0-dev.0",
+  "description": "MongoDB Shell E2E Tests Package",
+  "homepage": "https://github.com/mongodb-js/mongosh",
+  "author": "Compass Team <compass@mongodb.com>",
+  "manufacturer": "MongoDB Inc.",
+  "bin": {
+    "mongosh": "bin/mongosh.js"
+  },
```
Comment (addaleax) [inline packages/e2e-tests/package.json:10]:
```suggestion
```

File: packages/e2e-tests/package.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  "config": {
+    "unsafe-perm": true
+  },
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/mongodb-js/mongosh.git"
+  },
+  "scripts": {
+    "start": "node bin/mongosh.js",
```
Comment (addaleax) [inline packages/e2e-tests/package.json:19]:
```suggestion
```

File: packages/e2e-tests/package.json [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    "test-e2e": "cross-env TS_NODE_PROJECT=../../configs/tsconfig-mongosh/tsconfig.test.json mocha -r \"../../scripts/import-expansions.js\" --timeout 15000 --colors -r ts-node/register \"./test/e2e*.spec.ts\"",
+    "test-e2e-ci": "cross-env TS_NODE_PROJECT=../../configs/tsconfig-mongosh/tsconfig.test.json mocha -r \"../../scripts/import-expansions.js\" --timeout 15000 --colors -r ts-node/register \"./test/e2e*.spec.ts\"",
+    "eslint": "eslint",
+    "lint": "npm run eslint . && npm run prettier -- --check .",
+    "check": "npm run lint && npm run depcheck",
+    "depcheck": "depcheck",
+    "prepublish": "npm run compile",
+    "webpack-build": "npm run compile && webpack --mode production",
+    "webpack-build-dev": "npm run compile && webpack --mode development",
```
Comment (addaleax) [inline packages/e2e-tests/package.json:28]:
```suggestion
```

File: packages/e2e-tests/package.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        "darwin"
+      ],
+      "win-export-certificate-and-key": [
+        "win32"
+      ],
+      "get-console-process-list": [
+        "win32"
+      ]
+    }
+  },
```
Comment (addaleax) [inline packages/e2e-tests/package.json:51]:
```suggestion
```

File: packages/e2e-tests/package.json [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    "node-fetch": "^2.6.1",
+    "prettier": "^2.8.8",
+    "rimraf": "^3.0.2",
+    "webpack-merge": "^5.8.0"
+  },
+  "optionalDependencies": {
+    "get-console-process-list": "^1.0.4",
+    "macos-export-certificate-and-key": "^1.1.1",
+    "mongodb-crypt-library-version": "^1.0.3",
+    "win-export-certificate-and-key": "^1.1.1"
```
Comment (addaleax) [inline packages/e2e-tests/package.json:81]:
```suggestion
    "@mongosh/cli-repl": "0.0.0-dev.0",
```

File: packages/e2e-tests/webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+'use strict';
```
Comment (addaleax) [inline packages/e2e-tests/webpack.config.js:1]: Delete this file?

File: packages/e2e-tests/test/e2e.spec.ts
Comment (addaleax) [inline packages/e2e-tests/test/e2e.spec.ts:1]: What happened to `test-shell.ts` and `repl-helpers.ts`? You’ll still need to make the e2e tests actually _consume_ the mongosh that was compiled in the `cli-repl` package

**Review by addaleax — COMMENTED** [review]
I started reviewing this but began to realize that this is probably more of a draft PR than ready-for-review, sorry if this is too noisy

File: packages/e2e-tests/bin/mongosh.js (hunk 2 above)
Comment (lerouxb) [reply packages/e2e-tests/bin/mongosh.js:2]: I checked and it seems to be used from inside the test folder: https://github.com/mongodb-js/mongosh/blob/eddb8fb84b4aad6158c4a4a2b41e8cc9943735c6/packages/cli-repl/test/test-shell.ts#L70

File: packages/e2e-tests/test/e2e.spec.ts [hunk 9]
```diff
@@ -886,6 +890,10 @@ describe('e2e', function () {
       it('interrupts load()', async function () {
         const filename = path.resolve(
           __dirname,
+          '..',
+          '..',
+          'cli-repl',
+          'test',
           'fixtures',
           'load',
           'infinite-loop.js'
```
Comment (addaleax) [inline packages/e2e-tests/test/e2e.spec.ts:899]: A bunch of these fixtures are only used in e2e tests, we can move those, right?

**Review by addaleax — APPROVED** [review]

File: packages/e2e-tests/test/e2e.spec.ts (hunk 9 above)
Comment (lerouxb) [reply packages/e2e-tests/test/e2e.spec.ts]: My audit:

```
e2e:

load/hello1.js (both)
load/long-sleep.js
load/infinite-loop.js
load/throw.js (both)
exampleplayground.js
asi-script.js
require-base/
node-path/
curl.js

cli-repl:

fake-os-ca-provider.js
nodescript.js
load/a.js
load/b.js
load/c.js
load/hello1.js (both)
load/hello2.js
load/throw.js (both)
load/insertintotest.js
load/printisinteractive.js
```

**Comment by lerouxb** [comment]
evergreen refresh

**Pull request merged**